AGS_Engine/Compute/Extract/ExtractOne.cs

```diff
@@ -0,0 +1,92 @@
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Adapters.AGS;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Adapters.AGS
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Extracts the value with the highest score when comparing the query to the choices. The method uses the weighted ratio and full process.")]
+        [Input("query", "The string to carry out the fuzzy matching on.")]
+        [Input("choices", "A list of strings to compare the query against.")]
+        [Output("result", "A FuzzyStringResult containing the string, score and index resulting from the fuzzy matching algorithm.")]
+        public static FuzzyStringResult ExtractOne(string query, IEnumerable<string> choices)
+        {
+            return ExtractOne(query, choices, Scorer.DefaultRatioScorer);
+        }
+
+        /***************************************************/
+
+        [Description("Extracts the value with the highest score when comparing the query to the choices.")]
+        [Input("query", "The string to carry out the fuzzy matching on.")]
+        [Input("choices", "A list of strings to compare the query against.")]
+        [Input("scorer", "The method to use to score the strings when compared.")]
+        [Output("result", "A FuzzyStringResult containing the string, score and index resulting from the fuzzy matching algorithm.")]
+        public static FuzzyStringResult ExtractOne(string query, IEnumerable<string> choices, Scorer scorer = Scorer.DefaultRatioScorer)
```

Comment:
The first method should be deleted as it will cause problems when a call like `ExtractOne("something", choices)` is made. They are two methods to choose from that match the requirements.